src/id3_variants_training/ga4gh_API.py

```diff
@@ -86,18 +86,26 @@ def query_variants(self, chrom, start, end):
         Returns: 
             variant_name_list (list): list of variant names formatted in the for of `CHR:POS`
         """
-        variant_list = []
         req_body = {
             'datasetId' : self.dataset_id,
-            'start': start,
-            'end': end,
-            'referenceName': chrom
+            'logic': {'id': 'A'},
+            'components': [{'id': 'A', 'patients': {}}],
+            'results': [{
+                'start': start,
+                'end': end,
+                'referenceName': chrom,
+                'table': 'variants',
+                'fields': [
+                    'start',
+                    'end'
+                ]
+            }]
         }
-        r = requests.post('%s%s' %  (self.host_url, 'variants/search'), json=req_body).json()
+        r = requests.post('%s%s' %  (self.host_url, 'search'), json=req_body).json()
         if 'results' in r:
-            for variant in r['results']['variants']:
-                variant_list.append(':'.join([chrom, variant['start'], variant['end']]))
-        return variant_list
+            variant_set = {':'.join([chrom, variant['start'], variant['end']]) 
```

Comment:
Instead of making this a dictionary, you can convert it to a list and directly return the list. There may be duplicates since it's not being converted from a set to a list but in fetch_variants we already handle deduplicating it so there still shouldn't be duplicates in the end.

Reply:
It's actually a set, rather than a dictionary; I want it to deduplicate the variants (so we don't create a list, create a set from the list, then create a list from the set)

Reply:
If I do this here I can probably remove the deduplication from ID3_Class

Reply:
Either way is fine; it just seems like there's unnecessary work being done since we keep converting lists to sets and back again to deduplicate it but we don't need it everywhere.